zulipterminal/core.py

```diff
@@ -424,6 +426,8 @@ def open_in_browser(self, url: str) -> None:
                     [f"The link was successfully opened using {browser_name}"]
                 )
         except webbrowser.Error as e:
+            if isinstance(term, str):
+                os.environ["TERM"] = term  # resetting
```

Comment:
This certainly shouldn't be in the exceptional block. As it is now, TERM will not be reset if there is no error. If you want something to happen in every case, you could use `finally`.

The simplest solution would be to reset TERM as soon as the browser is determined. If so, this may be clearer with two try blocks, so you can isolate the 'only-terminal-browsers' error from others.

That is, unless the webbrowser module does other detection of this kind after that point, which could require the environment to be set for later operations?

Reply:
Using `finally` is a very good idea! 
Re using 2 try blocks, we don't need that since `TERM` environ isn't used anywhere during or after unsetting it. Resetting it in `finally` seems better it seems like the cleaner approach.